wordpress-mobile/WordPress-Android
# Issue/13135 remove location settings

Fixes #13135 

Geotagging is n[o longer supported](https://wordpress.com/support/replacing-the-older-wordpress-com-editor-with-the-wordpress-block-editor/) in new editors, so we are removing it from Post Settings.

I believe that the now deprecated old Places SDK was only used in post settings, so I removed the dependency and API key reference as well.

To test:
- Navigate to Post Settings.
- Notice that the Location setting option is gone.

PR submission checklist:
